lib/matplotlib/cbook.py

```diff
@@ -2487,6 +2490,41 @@ def pts_to_midstep(x, *args):
                    'step-post': pts_to_poststep,
                    'step-mid': pts_to_midstep}
 
+
+def get_index_y(y):
```

Comment:
@OceanWolf It does 'the right thing' with `pd.Series`.  This is also a place holder where we can put more brains to deal with other types of labeled data that knows what it's index is (and is not accessible via `obj.index`).  Eventually we should probably make this pluggable (like the units framework).